backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -1559,11 +1563,43 @@ export class ExperimentAssignmentService {
     return;
   }
 
+  private async getNewExperimentConditionAssignment(
+    experiment: Experiment,
+    user: ExperimentUser,
+    logger: UpgradeLogger,
+    enrollmentCount?: { conditionId: string; userCount: number }[]
+  ): Promise<ExperimentCondition> {
+    const isMoocletExperiment = this.moocletExperimentService.isMoocletExperiment(experiment.assignmentAlgorithm);
+
+    if (isMoocletExperiment && !env.mooclets.enabled) {
+      logger.error({
+        message: 'Mooclet experiment algorithm is indicated but mooclets are not enabled',
+        experiment,
+        user,
+      });
+      return undefined;
+    }
+
+    if (isMoocletExperiment && env.mooclets.enabled) {
+      return this.getConditionFromMoocletProxy(experiment, user);
+    } else {
+      return this.assignRandom(experiment, user, enrollmentCount);
+    }
+  }
+
+  private async getConditionFromMoocletProxy(experiment: Experiment, user: ExperimentUser) {
+    const userId = user.id;
+
+    const condition = await this.moocletExperimentService.getConditionFromMoocletProxy(experiment, userId);
```

Comment:
Do we want to create this variable just to return it?

Reply:
we don't have to, I guess I like to be explicit vs saving a couple of characters... it's also slightly easier to debug later if need be, all the data is assembled there in the local scope to spy or log out.